### ISSUE TITLE
Add length normalized loss metrics in API

### ISSUE THREAD

**Issue by liuzh91** [dmlc/gluon-nlp#1094]
## Description
In typical machine translation tasks, training/validation metric loss is computed using some loss function normalized by the length of target sequence. For example, in `train_gnmt.py`, the metric is computed with the following code:

```python
loss = loss_function(out, tgt_seq[:, 1:], tgt_valid_length - 1).mean()
loss = loss * (tgt_seq.shape[1] - 1) / (tgt_valid_length - 1).mean()
```
Current Mxnet `metric.loss` does not support length normalization. It will be great to add length normalized metric in the api.

## References
- https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/metric.py#L1661
- https://github.com/dmlc/gluon-nlp/blob/master/scripts/machine_translation/train_gnmt.py
